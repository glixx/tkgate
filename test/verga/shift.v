module top;
	reg [127: 0] big;
	reg [31:0] i;
	integer j;

	initial
		begin
			$display("Starting shift test");

			for (j=0; j<64; j = j + 1)
				begin
					$display("1 << %d == %08h", j, 1<<j);
				end
			
			i = 32'hdeadbeef;
			for (j=0; j<8; j = j + 1)
				begin
					$display("i = %08h", i);
					i = i << 4;
				end

			i = 32'hdeadbeef;
			for (j=0; j<8; j = j + 1)
				begin
					$display("i = %08h", i);
					i = i >> 4;
				end

			big = 128'h0123456789abcdefdeadbeefcafebabe;
	
			$display("big<<4   = %032h", big<<4);
			$display("big<<32  = %032h", big<<32);
			$display("big<<39  = %032h", big<<39);
			$display("big<<64  = %032h", big<<64);
			$display("big<<68  = %032h", big<<68);
			$display("big<<200 = %032h", big<<200);

			$display("big>>4   = %032h", big>>4);
			$display("big>>32  = %032h", big>>32);
			$display("big>>39  = %032h", big>>39);
			$display("big>>64  = %032h", big>>64);
			$display("big>>68  = %032h", big>>68);
			$display("big>>200 = %032h", big>>200);

		end

endmodule
