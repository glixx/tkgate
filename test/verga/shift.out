Starting shift test
1 << 0 == 00000001
1 << 1 == 00000002
1 << 2 == 00000004
1 << 3 == 00000008
1 << 4 == 00000010
1 << 5 == 00000020
1 << 6 == 00000040
1 << 7 == 00000080
1 << 8 == 00000100
1 << 9 == 00000200
1 << 10 == 00000400
1 << 11 == 00000800
1 << 12 == 00001000
1 << 13 == 00002000
1 << 14 == 00004000
1 << 15 == 00008000
1 << 16 == 00010000
1 << 17 == 00020000
1 << 18 == 00040000
1 << 19 == 00080000
1 << 20 == 00100000
1 << 21 == 00200000
1 << 22 == 00400000
1 << 23 == 00800000
1 << 24 == 01000000
1 << 25 == 02000000
1 << 26 == 04000000
1 << 27 == 08000000
1 << 28 == 10000000
1 << 29 == 20000000
1 << 30 == 40000000
1 << 31 == 80000000
1 << 32 == 00000000
1 << 33 == 00000000
1 << 34 == 00000000
1 << 35 == 00000000
1 << 36 == 00000000
1 << 37 == 00000000
1 << 38 == 00000000
1 << 39 == 00000000
1 << 40 == 00000000
1 << 41 == 00000000
1 << 42 == 00000000
1 << 43 == 00000000
1 << 44 == 00000000
1 << 45 == 00000000
1 << 46 == 00000000
1 << 47 == 00000000
1 << 48 == 00000000
1 << 49 == 00000000
1 << 50 == 00000000
1 << 51 == 00000000
1 << 52 == 00000000
1 << 53 == 00000000
1 << 54 == 00000000
1 << 55 == 00000000
1 << 56 == 00000000
1 << 57 == 00000000
1 << 58 == 00000000
1 << 59 == 00000000
1 << 60 == 00000000
1 << 61 == 00000000
1 << 62 == 00000000
1 << 63 == 00000000
i = deadbeef
i = eadbeef0
i = adbeef00
i = dbeef000
i = beef0000
i = eef00000
i = ef000000
i = f0000000
i = deadbeef
i = 0deadbee
i = 00deadbe
i = 000deadb
i = 0000dead
i = 00000dea
i = 000000de
i = 0000000d
big<<4   = 123456789abcdefdeadbeefcafebabe0
big<<32  = 89abcdefdeadbeefcafebabe00000000
big<<39  = d5e6f7ef56df77e57f5d5f0000000000
big<<64  = deadbeefcafebabe0000000000000000
big<<68  = eadbeefcafebabe00000000000000000
big<<200 = 00000000000000000000000000000000
big>>4   = 00123456789abcdefdeadbeefcafebab
big>>32  = 000000000123456789abcdefdeadbeef
big>>39  = 000000000002468acf13579bdfbd5b7d
big>>64  = 00000000000000000123456789abcdef
big>>68  = 000000000000000000123456789abcde
big>>200 = 00000000000000000000000000000000
